docs/USAGE-GUIDE.md

```diff
@@ -1658,6 +1723,7 @@ Once the output is set the asynchronous private key operation can be completed b
 following the steps outlined [above](#Asynchronous-private-key-operations-related-calls)
 to apply the operation and free the op object.
 
+The [handshake algorithm getters](###S2N-Get-Selected-Handshake-Algorithms) and [certificate context](#s2n\_cert\_chain\_and\_key\_get\_ctx) APIs can be helpful for determining the private key and what private key algorithm to use.
```

Comment:
This should probably be integrated into the above instructions rather than mentioned separately afterwards.

Also, you should probably be more specific on how the certificate context helps here. I believe the idea is to use it to keep track of a which private key is associated with the cert.